### PR TITLE
Ana/allow use any token to pay fee in Claim Rewards

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Now if there is no available balance of the staking denom users can still pay fees in Claim Rewards with any other token.

Second and last part of the splitted #3662 

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
This is ignoring that different tokens have different values and need different gas prices. In which situation is this PR needed?

**Comment by Bitcoinera** [comment]
> This is ignoring that different tokens have different values and need different gas prices. In which situation is this PR needed?

gasPrice comes from the API already 🤷‍♀ 

This situation:

**Comment by Bitcoinera** [comment]
Oops, what happened with netlify here xD ?

File: src/ActionModal/components/ModalWithdrawRewards.vue [hunk 1]
```diff
@@ -76,6 +78,20 @@ export default {
     },
     validatorsWithRewards() {
       return this.rewards.length > 0
+    },
+    feeDenom() {
+      // since it is cheaper to pay fees with the staking denom (at least in Tendermint), we return this denom
+      // as default if there is any available balance. Otherwise, we return the first balance over 0
+      // TODO: change to be preferrably the same token that is shown as claimed, although not so important
+      if (this.balances && this.balances.length > 0) {
+        return this.balances.filter(({ denom }) => denom === this.denom)[0] &&
```
Comment (faboweb) [inline src/ActionModal/components/ModalWithdrawRewards.vue]: refactor this and next line to only use `this.balances.filter(({ denom }) => denom === this.denom)` once

**Comment by faboweb** [comment]
ok, accepting the approach
does the UI show, that we using a different token for the fees?

**Comment by Bitcoinera** [comment]
> does the UI show, that we using a different token for the fees?

Yes, it does. As I told you in the meeting, now there is just a prop you pass to ActionModal, selectedDenom, and that does all the magic

**Comment by faboweb** [comment]
Remember that a normal user doesn't see the gasPrice input. But yes the table shows the denom. We should show a warning that another token is used still. But another PR.

File: src/ActionModal/components/ModalWithdrawRewards.vue [hunk 2]
```diff
@@ -84,9 +84,10 @@ export default {
       // as default if there is any available balance. Otherwise, we return the first balance over 0
       // TODO: change to be preferrably the same token that is shown as claimed, although not so important
       if (this.balances && this.balances.length > 0) {
-        return this.balances.filter(({ denom }) => denom === this.denom)[0] &&
-          this.balances.filter(({ denom }) => denom === this.denom)[0].amount >
-            0
+        const stakingDenomBalance = this.balances.filter(
+          ({ denom }) => denom === this.denom
+        )[0]
```
Comment (faboweb) [inline src/ActionModal/components/ModalWithdrawRewards.vue]:
```suggestion
        const nunZeroBalances = this.balances.filter(
          ({ amount }) => amount > 0
        )
```

Comment (Bitcoinera) [reply src/ActionModal/components/ModalWithdrawRewards.vue]: I think I just applied this successfully

File: src/ActionModal/components/ModalWithdrawRewards.vue [hunk 3]
```diff
@@ -84,12 +84,13 @@ export default {
...
+        const stakingDenomBalance = nonZeroBalances.find(
           ({ denom }) => denom === this.denom
-        )[0]
-        return stakingDenomBalance && stakingDenomBalance.amount > 0
-          ? this.denom
-          : this.balances.filter(({ amount }) => amount > 0)[0].denom
+        )
+        return stakingDenomBalance
+          ? stakingDenomBalance.denom
+          : nonZeroBalances[0].denom
```
Comment (faboweb) [inline src/ActionModal/components/ModalWithdrawRewards.vue]:
```suggestion
          : nonZeroBalances.length > 0 ? nonZeroBalances[0].denom : this.denom
```

**Pull request merged**